fix: jdate.js and framework7 calendar component #1631

mer30hamid opened the pull request:
Hi, this changes are safe and will resolve this issue: 
https://github.com/framework7io/Framework7/issues/1631

mer30hamid committed:
fixed issue with framework7 calendar component #1631

mer30hamid committed:
fixed issue: framework7 calendar component #1631

tahajahangir commented:
Hi Hamid,
Please explain more about this PR. What's the issue you've tried to solve? Please provide examples or test-cases.
You renamed `Date` to `refDate`. What's the purpose for this change? (Prefer not changing it if not necessary)
There is some other white-space changes in this PR. Please eliminate unnecessary change.

mer30hamid commented:
Hi Taha,
I had to chnge the _Date_ to _refDate_  because there were conflict with [f7](https://framework7.io/) code, you can see it in action here:

https://jsfiddle.net/xfkx2nz0/8/

press f12 and click "my date" and you can see this error:
`jdate-class.js:104 Uncaught RangeError: Maximum call stack size exceeded`

![why-refdate](https://user-images.githubusercontent.com/1561497/34322099-8eac3bd4-e833-11e7-8369-96b8dea8955f.jpg)

I think it's better to change it, there is no error when using minimized version because there is no "Date" in minimized version, you can see it here : 
https://jsfiddle.net/xfkx2nz0/9/

but as you can see, this have the issue with date picker https://github.com/framework7io/framework7/issues/1631

so you asked 

> What's the issue you've tried to solve?

1. I want pass my parameters to JDate as "String", is its output identical when i pass parameters as "number" ?
answer is _NO_
 see it here : https://jsfiddle.net/xfkx2nz0/10/
in logic of the current code, if first parameter is "string" , then all other parameters are ignored and first parameter assumed as "Date String" so I've added "isNumeric()" function to check first parameter (a) to determine if it is number (not in type, but in fact !) then in next step I converted them all to number: 
```
if (typeof a == 'string') a = Number(a);
if (typeof month == 'string') month = Number(month);
if (typeof day == 'string') day = Number(day);
if (typeof hour == 'string') hour = Number(hour);
if (typeof minute == 'string') minute = Number(minute);
if (typeof second == 'string') second = Number(second);
if (typeof millisecond == 'string') millisecond = Number(millisecond);
```
see it with my changes : https://jsfiddle.net/xfkx2nz0/11/

2. I changed ` day || 1` to ` (typeof(day) !== "undefined" && day !== null)? day:1)` but why?

> The || operator in JavaScript doesn't necessarily return true or false. It's exact behavior is this:
> **If the first operand is truthy, it evaluates to the first operand. Otherwise, it evaluates to the second.**

read more [here +](https://stackoverflow.com/questions/40814333/why-does-false-null-return-null-while-null-false-returns-false/40814362)

because  the '0'  is not truthy !

compare this (current JDate) : https://jsfiddle.net/xfkx2nz0/12/
with this: (my forked JDate): https://jsfiddle.net/xfkx2nz0/13/

and you mentioned that 

> There is some other white-space changes in this PR. Please eliminate unnecessary change.

I even removed unnecessary spaces that already was there  !  around "createWrapper" , in this pic left code have my changes

![removed-spaces](https://user-images.githubusercontent.com/1561497/34322528-67430054-e83e-11e7-95c0-660446d20d32.jpg)

I changed `.split(' '),` to `.split(' ');` because I think its a syntactic fault.

but if you mean line breaks, there were only 2 and I removed them now :))

this is my first PR and there is other changes than "jdate-class.js", and they are not important, I minimized scripts using a batch file and this is appeared as "adding" in this PR too !!

sorry for my bad English, 
you can contact me by telegram, my id is mer30hamid,
I'm glad to be in touch with you.

mer30hamid committed:
Update minify.sh
